tinygrad/codegen/linearizer.py

```diff
@@ -310,9 +310,18 @@ def upcast_strides(buf:int):
       # end the late reduce loop
       self.load_cache.clear()
 
-      # all local indices which were used for group_for_reduce are not valid any more and should be replaced with fake NumNode(0), since they have
-      # been rewritten with fake end_local_idxs.
-    return (accs, loaded_buffers, fake_reduce_idxs, local_idxs[:self.local_dims] + [NumNode(0) for i in range(self.group_for_reduces)], upcast_idxs)
```

Comment:
I can't merge this because it breaks `test/test_linearizer_failures.py TestLinearizerFailures.test_failure_12`
you can see the kernel output with DEBUG=4 
fyi the original change was introduced in https://github.com/tinygrad/tinygrad/pull/3180 - Seems like it's not a well-tested edge case but will potentially break gpt2 on BEAM=2